src/goto-instrument/cover.cpp

```diff
@@ -1445,13 +1448,32 @@ bool instrument_cover_goals(
       c=coverage_criteriont::COVER;
     else
     {
-      msg.error() << "unknown coverage criterion" << messaget::eom;
+      msg.error() << "unknown coverage criterion "
+                  << criterion_string
```

Comment:
For extra clarity it might be good to place the criterion_string in single quotes in the output.